net/test/node_bulk.c

```diff
@@ -1277,17 +1284,23 @@ static void node_bulk_worker(struct node_bulk_ctx *ctx)
 			m0_net_buffer_event_notify(ctx->nbc_net.ntc_tm,
 						   &tm_chan);
 		}
+	#ifndef ENABLE_LIBFAB
 		ctx->nbc_callback_executed = false;
+	#endif
 		/* execute network buffer callbacks in this thread context */
 		m0_net_buffer_event_deliver_all(ctx->nbc_net.ntc_tm);
 		M0_ASSERT(ergo(pending, ctx->nbc_callback_executed));
+	#ifndef ENABLE_LIBFAB
 		/* state transitions from final states */
 		node_bulk_state_transition_auto_all(ctx);
+	#endif
 		/* update copy of statistics */
 		m0_net_test_nh_sd_copy_locked(&ctx->nbc_nh);
+	#ifndef ENABLE_LIBFAB
 		/* wait for STOP command or buffer event */
 		if (!ctx->nbc_callback_executed)
 			m0_chan_wait(&ctx->nbc_stop_clink);
+	#endif
```

Comment:
That's very difficult to follow. Please create a helper function:
```
static void net_buf_worker_cb(...)
{
#ifndef ENABLE_LIBFAB
		ctx->nbc_callback_executed = false;
	#endif
		/* execute network buffer callbacks in this thread context */
		m0_net_buffer_event_deliver_all(ctx->nbc_net.ntc_tm);
		M0_ASSERT(ergo(pending, ctx->nbc_callback_executed));
	#ifndef ENABLE_LIBFAB
		/* state transitions from final states */
		node_bulk_state_transition_auto_all(ctx);
	#endif
		/* update copy of statistics */
		m0_net_test_nh_sd_copy_locked(&ctx->nbc_nh);
	#ifndef ENABLE_LIBFAB
		/* wait for STOP command or buffer event */
		if (!ctx->nbc_callback_executed)
			m0_chan_wait(&ctx->nbc_stop_clink);
	#endif
}
```
Call it from node_bulk_worker(), thencreate 2 versions of node_bulk_worker_cb() with a single #ifdef. Also all a comment with explanation.

Also, in addition to ENABLE_LIBFAB, define another macro USE_LIBFAB, which is _always_ defined to 0 (no libfab) or 1 (use libfab). Instead of `#ifdef ENABLE_LIBFAB`, use `if (USE_LIBFAB)` where possible.

Reply:
Done.